Add script to decontaminate datasets against benchmark datasets (#416)
huggingface/open-r1: README.md

    --config recipes/Qwen2.5-1.5B-Instruct/grpo/config_demo_code.yaml
```

#### Data decontamination

Following [s1: Simple test-time scaling](https://arxiv.org/abs/2501.19393) the data can be decontaminated using the script at: [scripts/decontaminate.py](./scripts/decontaminate.py), which decontaminates a dataset using 8-grams and deduplicate the data. Sample run:

```shell
python scripts/decontaminate.py \

>     --dataset "open-r1/verifiable-coding-problems-python-10k" \

Reviewer: let's use the full dataset, since the 10k one is a bit buggy
```suggestion
    --dataset "open-r1/verifiable-coding-problems-python" \
```

    --problem_column problem \

>     -c

Reviewer: This is a bit cryptic, so let's use the full name:

```suggestion
    --cleanup
```

```

It will decontaminate against the benchmark datasets, and remove the contaminated samples afterwards. If no argument `--new_dataset_name` is provided, the same dataset will be reused, adding a `_decontaminated`. It runs against the prompt, which for this dataset is the column `problem`, but a different one can be provided.

Arguments for the script:

```shell
usage: decontaminate.py [-h] --dataset DATASET [--split SPLIT] [--ngram_size NGRAM_SIZE] [--problem_column PROBLEM_COLUMN] [--cleanup] [--new_dataset_name NEW_DATASET_NAME]

options:
  -h, --help            show this help message and exit
  --dataset DATASET     Name of the dataset to check for contamination.
  --split SPLIT         Split to check for contamination, defaults to `train`.
  --ngram_size NGRAM_SIZE
                        Size of n-grams to build, defaults to 8.
  --problem_column PROBLEM_COLUMN
                        Name of the column containing the problem (prompt).

>   --cleanup, -c         Wether to remove the contaminated rows before pushing the dataset.

Reviewer: Let's keep it consistent:
```suggestion
  --cleanup           Whether to remove the contaminated rows before pushing the dataset.
```
